projects/rocr-runtime/runtime/hsa-runtime/core/runtime/runtime.cpp

```diff
@@ -3841,7 +3843,12 @@ Runtime::MappedHandle::MappedHandle(MemoryHandle *mem_handle, AddressHandle *add
     shareable_handle(shareable_handle)
 {
   /* Create a CPU mapping with PROT_NONE */
+  #if defined(__linux__)
```

Comment:
We added PROT_NONE CPU mapping for an RDMA feature. This was done so that kernel-mode drivers on Linux could then look-up that virtual address, otherwise there would be no entry in kernel mode for that memory allocation.

I think Linux implementation could be different and probably does not need this, so I would suggest you put this whole section that creates the default CPU mapping inside #ifdef linux.

Mapping to the GPU instead is not correct as there will be a duplicate call to map to GPU.